aten/src/ATen/native/SpectralOps.cpp

```diff
@@ -561,13 +561,19 @@ void _cufft_clear_plan_cache(int64_t device_index) {
 }
 
 Tensor fft(const Tensor& self, const int64_t signal_ndim, const bool normalized) {
+  TORCH_WARN_ONCE(
+    "torch.fft is deprecated and will be removed in pytorch 1.8. "
```

Comment:
Reviewing this now because I'd like to get this in for 1.7 (branch cut next week Tuesday) so `torch.fft()` is deprecated in 1.7 and can be removed in 1.8. Realistically this PR and the one it's stacked will need to be done and landable before Monday to make that happen. Let me know if there's something I can do to help, @peterbell10. 

Overall this looks good but it needs two things:

- the _tensor_docs.py for `torch.fft` also need a (deprecation) warning with language like this (except it can link to the torch.fft module documentation)
- this warning should tell people to import torch.fft to get access to the functions that are replacing `torch.fft()`

Maybe something like:

"The function torch.fft is deprecated and will be removed in a future PyTorch release. Use the new torch.fft module functions, instead, by importing torch.fft and calling torch.fft.fft or torch.fft.fftn." 

?

Reply:
@mruberry I've updated the docs and warning message. Is there a policy on giving the version a deprecated function will be removed? I've put `torch.fft()` as expiring in 1.8 but left the others as "a furture PyTorch release" since that's not a priority.

Reply:
No policy. "In a future PyTorch release" gives us some latitude.